src/base/async/int.c

```diff
@@ -1898,8 +1898,8 @@ static int redir_it(void)
   LWORD(esi) = sda_lo; SREG(ds) = sda_hi;
   LWORD(ebx) = DOS_SUBHELPER_MFS_REDIR_INIT;
   LWORD(eax) = DOS_HELPER_MFS_HELPER;
-  if (mfs_inte6() == TRUE) {  /* Do we have a functioning redirector? */
-    redirect_devices();
+  if (mfs_inte6() == TRUE && LWORD(eax)) {
```

Comment:
mfs_inte6() always returns TRUE to indicate the helper handled the function, the value in AX is the value to be returned to the caller. With the init call the value returned is 0 or 1, essentially indicating whether we have valid info for using the redirector. Previously we didn't check that we have valid CDS info with the internal redirector initialisation call, but since we do it with emufs.sys I think it makes sense to do it here also. I suppose you could have LWORD(eax) == TRUE, but it seems a little pointless.